core/src/test/java/org/infinispan/marshall/DefensiveCopyTest.java

```diff
@@ -50,7 +51,7 @@ public void testSafetyAfterRetrieving() {
       Person cachedPerson = this.<Integer, Person>cache().get(k);
       assertEquals(person, cachedPerson);
       cachedPerson.setName("Mr Digweed");
-      assertEquals(new Person("Mr Coe"), cache.get(k));
+      assertEquals(person, cache.get(k));
```

Comment:
This is not right; if `person == cachedPerson` and Infinispan did not make a defensive copy, this would succceed.

Reply:
If I understand tests correctly, this is tested in previous `testOriginalReferenceSafety`, this one should test that modification to instance obtained from cache doesn't affect instance in the cache and thus creating new object in assert is redundant.